pkg/action/custom_resource_lister.go

```diff
@@ -0,0 +1,170 @@
+		Name: fmt.Sprintf("%s.%s", o.PackageName, o.config.Namespace),
+	}
+
+	result, err := o.listAll(ctx, opKey)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+// FindOperator finds an operator object on-cluster provided a package and namespace.
+func (o *OperatorListOperands) findOperator(ctx context.Context, key types.NamespacedName) (*v2alpha1.Operator, error) {
+	operator := v2alpha1.Operator{}
+
+	err := o.config.Client.Get(ctx, key, &operator)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil, fmt.Errorf("package %s not found", key.Name)
+		}
+		return nil, err
+	}
+	return &operator, nil
+}
+
+// Unzip finds the CSV referenced by the provided operator and then inspects the spec.customresourcedefinitions.owned
+// section of the CSV to return a list of APIs that are owned by the CSV.
+func (o *OperatorListOperands) unzip(ctx context.Context, operator *v2alpha1.Operator) ([]v1alpha1.CRDDescription, error) {
+	csv := v1alpha1.ClusterServiceVersion{}
+	csvKey := types.NamespacedName{}
+
+	if operator.Status.Components == nil {
+		return nil, fmt.Errorf("could not find underlying components for operator %s", operator.Name)
+	}
+	for _, resource := range operator.Status.Components.Refs {
+		if resource.Kind == v1alpha1.ClusterServiceVersionKind {
+			csvKey.Name = resource.Name
+			csvKey.Namespace = resource.Namespace
+			break
+		}
+	}
+
+	if csvKey.Name == "" && csvKey.Namespace == "" {
+		return nil, fmt.Errorf("could not find underlying CSV for operator %s", operator.Name)
+	}
+
+	err := o.config.Client.Get(ctx, csvKey, &csv)
+	if err != nil {
+		return nil, fmt.Errorf("could not get %s CSV on cluster: %s", csvKey.String(), err)
+	}
+
+	// check if owned CRDs are defined on the csv
+	if len(csv.Spec.CustomResourceDefinitions.Owned) == 0 {
+		return nil, fmt.Errorf("no owned CustomResourceDefinitions specified on CSV %s, no custom resources to display", csvKey.String())
+	}
+
+	// check CSV is not in a failed state (to ensure some OLM multitenancy rules are not violated)
+	if csv.Status.Phase != v1alpha1.CSVPhaseSucceeded {
+		return nil, fmt.Errorf("CSV underlying operator is not in a succeeded state")
+	}
+
+	return csv.Spec.CustomResourceDefinitions.Owned, nil
+}
+
+// List takes in a CRD description and finds the associated CRs on-cluster.
+// List can return a potentially unbounded list that callers may need to paginate.
+func (o *OperatorListOperands) list(ctx context.Context, crdDesc v1alpha1.CRDDescription, namespaces []string) (*unstructured.UnstructuredList, error) {
+	result := &unstructured.UnstructuredList{}
+
+	list := unstructured.UnstructuredList{}
+	gvk := schema.GroupVersionKind{
+		Group:   crdDesc.Name,
+		Version: crdDesc.Version,
+		Kind:    crdDesc.Kind,
+	}
+	result.SetGroupVersionKind(gvk)
+	if err := o.config.Client.List(ctx, &list); err != nil {
+		return nil, err
+	}
+
+	// trim down CRs in list to match target namespaces
+	if namespaces == nil {
```

Comment:
This will account for `nil` or empty.
```suggestion
	if len(namespaces) == 0 {
```